### ISSUE TITLE
Not possible to use a custom exclusion strategy with JMSSerializer

### ISSUE THREAD

**Issue by TobiasXy** [FriendsOfSymfony/FOSRestBundle#1536]
In my project I wrote a custom exclusion strategy for the JMSSerializer.
In version 1.7 I added it like this:

```
$context = SerializationContext::create()
            ->addExclusionStrategy($this->get('my_exclusion_strategy'));
$view->setSerializationContext($context);
```

Now in version 2.0 there is the new (generic) Context class, but it seems there is no way to set a custom exclusion strategy. Also there is apparently no way to hook into the JMSSerializerAdapter, (which is creating the actual SerializationContext of the JMSSerializer) to add the exclusion strategy there.

Do I really have to override the JMSSerializerAdapter to add a custom exclusion strategy or did I miss something obvious?

**Comment by Ener-Getick** [comment]
What about using a new attribute ``exclusion_strategies`` ?

cc @xabbuh

**Comment by TobiasXy** [comment]
Exclusion strategies are specific to the JMSSerializer, right?
Adding a property `exclusion_strategies` to the generic Context class would make it somehow specific again or did I misunderstand you there?

**Comment by Ener-Getick** [comment]
It would only be used by the jms serializer but i don't see another solution.
Except maybe making the adapters extandable.

**Comment by xabbuh** [comment]
Sounds like a good idea. A comment may be needed to explain that this might not be supported by adapters (or we could make this comment even more generic so that it applies to the `Context` class as a whole).

**Comment by alanhartless** [comment]
In the same boat here - ended up just downgrading to 1.8 after realizing this would be an issue as I'm also using a JMS custom exclusion strategy.

**Comment by Ener-Getick** [comment]
Can someone submit a PR for this one ? I already have a lot of PRs to manage :/

**Comment by podorozhny** [comment]
+1

**Comment by bblue** [comment]
Is anyone working on a PR for this? I'll give it a shot if not...

**Comment by Ener-Getick** [comment]
I don't think so. I think updating the jms adapter should be enough.

**Comment by xabbuh** [comment]
This was implemented in #1583.
